### ISSUE TITLE
Apply modules occurred 404 error

### ISSUE THREAD

**Issue by JayChanggithub** [dynatrace-oss/terraform-provider-dynatrace#248]
Hi Team, 

We encountered issue regarding apply dynatrace modules occured 404 error in some environments. 
These symptoms that we guess caused by some manually configuration previously. have any possible to fix massive these issue effecient way to do?  The specific logs error like `Error: {
dynatrace-tf_1  | │   "code": 404,
dynatrace-tf_1  | │   "message": "failed to resolve tenant \u003ca634a686-1515-433f-bb92-acc7e1df0f16\u003e",
dynatrace-tf_1  | │   "constraintViolations": null
dynatrace-tf_1  | │ }
`

terraform revision: `Terraform v0.15.3`
dynatrace dashboard: `1.262.152.20230324-133703`

```
terraform {
    required_providers {
        dynatrace   = {
            version = "~>1.11.0"
            source  = "dynatrace-oss/dynatrace"
        }
        azurerm = {
            version = "~>2.0"
        }
    }
```

**Comment by Dynatrace-Reinhard-Pilz** [comment]
Hello Jay,

I would have to see the configuration section of your provider to tell, what's going wrong exactly.
Based on the error message above, it SHOULD look like this
```
provider "dynatrace" {
  dt_env_url   = "https://DYNATRACE_HOST/e/a634a686-1515-433f-bb92-acc7e1df0f16"
  dt_api_token = "dt0c01.########################.################################################################"  
}
```
or alternatively you can configure the provider using these environment variables
```
DYNATRACE_ENV_URL=https://DYNATRACE_HOST/e/a634a686-1515-433f-bb92-acc7e1df0f16
DYNATRACE_API_TOKEN=dt0c01.########################.################################################################
```

What happens, when you enter `https://DYNATRACE_HOST/e/a634a686-1515-433f-bb92-acc7e1df0f16/api/config/v1/anomalyDetection/hosts` (after replacing `DYNATRACE_HOST` with the IP address or server name of your Dynatrace cluster) in the address line of your browser? Are you getting the same error message?

Is it possible, that you have multiple Dynatrace Clusters and you're simply addressing the wrong cluster for the environment `a634a686-1515-433f-bb92-acc7e1df0f16`?

**Comment by JayChanggithub** [comment]
@Dynatrace-Reinhard-Pilz  That's okayed let me try again. Also i had been adopted `-parallelism=1`, when executing terraform apply. However still encountered 404 error in these environments. The another enviroments are fine without any error. Using the same code to do.

**Comment by JayChanggithub** [comment]
@Dynatrace-Reinhard-Pilz 

I fixed issue on my end. After adopted `-parallelism=1` also upgrade provider to `1.28.0` the 404 error not occurred so far.
